mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -2543,6 +2543,17 @@ def start_trace(self, trace_info: "TraceInfo") -> TraceInfo:
                 # Trace already exists (likely created by log_spans())
                 # Use merge to update with start_trace() data, preserving any logged spans
                 session.rollback()
+                db_sql_trace_info = (
+                    session.query(SqlTraceInfo)
+                    .filter(SqlTraceInfo.request_id == trace_id)
+                    .one_or_none()
+                )
+                if db_sql_trace_info:
+                    for tag in db_sql_trace_info.tags:
+                        if tag.key not in tags:
+                            sql_trace_info.tags.append(
+                                SqlTraceTag(request_id=trace_id, key=tag.key, value=tag.value)
+                            )
```

Comment:
Was something broken before? What case is this covering?

Reply:
This is added because I added `SqlTraceTag(key=TraceTagKey.SPANS_LOCATION, value=TRACKING_STORE, request_id=trace_id)` in log_spans to ensure we know the spans are stored in the Spans table, and when we dual write (with log_spans and export_traces) start_trace also writes trace_info, so we need to make sure the tags are consistent between them.
Without this change, there are tests failing due to `AssertionError: Dependency rule on column 'trace_info.request_id' tried to blank-out primary key column 'trace_tags.request_id' on instance '<SqlTraceTag at 0x179716380>'`